docs/samples/python/langextract/index.md

```diff
@@ -0,0 +1,116 @@
+
+## Introduction
+
+Presidio supports language model-based PII/PHI detection for flexible entity recognition using language models (LLMs, SLMs, etc.). This approach enables detection of both:
+- **PII (Personally Identifiable Information)**: Names, emails, phone numbers, SSN, credit cards, etc.
+- **PHI (Protected Health Information)**: Medical records, health identifiers, etc.
+
+The current implementation uses [LangExtract](https://github.com/google/langextract) with **Ollama** for local model deployment. Additional provider integrations will be added soon.
+
+## Entity Detection Capabilities
+
+Unlike pattern-based recognizers, language model-based detection is flexible and depends on:
+- The language model being used
+- The prompt description provided
+- The few-shot examples configured
+
+The default configuration includes examples for common PII/PHI entities such as PERSON, EMAIL_ADDRESS, PHONE_NUMBER, US_SSN, CREDIT_CARD, MEDICAL_LICENSE, and more. 
+**You can customize the prompts and examples to detect any entity types relevant to your use case**.
+
+For the default entity mappings and examples, see the [default configuration](https://github.com/microsoft/presidio/blob/main/presidio-analyzer/presidio_analyzer/conf/langextract_config_ollama.yaml).
+
+## Prerequisites
+
+### Setting up Ollama
+
+You have two options to set up Ollama:
+
+**Option 1: Docker Compose** (recommended)
+```bash
+# Start Ollama service
+docker compose up -d ollama
+
+# Pull the language model (required - takes ~1-2 minutes)
+docker exec -it presidio-ollama-1 ollama pull gemma2:2b
+```
+
+**Option 2: Manual setup**
+Follow the [official LangExtract Ollama guide](https://github.com/google/langextract?tab=readme-ov-file#using-local-llms-with-ollama).
+
+!!! note "Note"
+    The model must be pulled before using the recognizer. The default model is `gemma2:2b` (~1.6GB).
+
+## Language Model-based Recognizer Implementation
+
+Presidio provides a hierarchy of recognizers for language model-based PII/PHI detection:
+
+- **`LMRecognizer`**: Abstract base class for all language model recognizers (LLMs, SLMs, etc.)
+- **`LangExtractRecognizer`**: Abstract base class for LangExtract library integration (model-agnostic)
+- **`OllamaLangExtractRecognizer`**: Concrete implementation for Ollama local language models
+
+[The implementation can be found here](https://github.com/microsoft/presidio/blob/main/presidio-analyzer/presidio_analyzer/predefined_recognizers/third_party/ollama_langextract_recognizer.py).
+
+## How to integrate Language Model-based detection into Presidio
+
+### Option 1: Enable in Configuration (Recommended)
+
+1. Install with langextract support and set up Ollama (see Prerequisites above):
+   ```sh
+   pip install presidio-analyzer[langextract]
+   ```
+
+2. Enable the recognizer in `default_recognizers.yaml`:
+   ```yaml
+   - name: OllamaLangExtractRecognizer
+     enabled: true  # Change from false to true
+   ```
```

Comment:
Consider being more explicit here - what should the user do after changing the yaml file? If the user pip installs presidio, they don't even know that there is this yaml file. If they clone, they still might be using the installed package and changing it would have no effect. My suggestion:
1. Add a link to default_recognizers.yaml on the repo (like you show with the custom yaml below)
2. Add a short python snippet that shows how to load this config with ollama.